### DISCUSSION TITLE
nginx ssl-terminating reverse proxy to foundryvtt-docker on a separate host

### DISCUSSION THREAD

**Discussion by uberDoward** [felddy/foundryvtt-docker#676] [Q&A] [answered]
I have to be missing something silly, here.  I've got a docker host on my network @ 192.168.2.210, and FoundryVTT works perfectly fine from http://192.168.2.210:30000.   My nginx reverse proxy, when accessing from https://foundry.dowardindustries.com, though, only serves up the background and the WSS connection fails.

nginx is on my primary incoming server @ 192.168.2.212, and I'm simply using nginx's reverse proxy to shuffle the traffic from foundry.dowardindustries.com over from 192.168.2.212 to the container @ 192.168.2.210:30000.   All works fine locally, but I get wss:// connection errors from https://foundry.dowardindustries.com... 

nginx configuration:
```
server {
    if ($host = foundry.dowardindustries.com) {
        return 301 https://$host$request_uri;
    } # managed by Certbot

     listen 80;
     server_name foundry.dowardindustries.com;
     return 301 https://$host$request_uri;
}

server {
    listen 443 http2 ssl;
    server_name foundry.dowardindustries.com;
    client_max_body_size 300M;

    location / {
        proxy_set_header Host $host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;

        # These are important to support WebSockets
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";

        proxy_pass http://192.168.2.210:30000;
    }

    ssl_certificate /etc/letsencrypt/live/plex.dowardindustries.com/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/plex.dowardindustries.com/privkey.pem; # managed by Certbot
}
```

options.json in the container:
```
FoundryVTT | 2023-05-04 03:23:01 | [info] Application Options:
{
  "awsConfig": null,
  "compressStatic": false,
  "fullscreen": false,
  "hostname": "foundry.dowardindustries.com",
  "language": "en.core",
  "localHostname": null,
  "passwordSalt": null,
  "port": 30000,
  "protocol": null,
  "proxyPort": 443,
  "proxySSL": true,
  "routePrefix": null,
  "sslCert": null,
  "sslKey": null,
  "updateChannel": "stable",
  "upnp": false,
  "upnpLeaseDuration": null,
  "world": null,
  "adminPassword": null,
  "serviceConfig": null
}
```

Here's the error:
```
can’t establish a connection to the server at wss://foundry.dowardindustries.com/socket.io/?session=2c17d5b791ded57daae4e340&EIO=4&transport=websocket.
```

Yet it works perfectly fine from http://192.168.2.210:30000 when accessed from my local network...

Note: I do have multiple other services running via this reverse proxy, to a smattering of other devices around the network, and this is the only one having issues, and I'm 99% certain it's the websocket connecting causing me grief...

Seems like I must be missing something basic, here.   Appreciate any help!

**Comment by uberDoward** [comment, marked as answer]
Completed this with help from reddit - turns out, this was all working excellent from external to my network.  I added a new A record for my local DNS so that foundry.dowardindustries.com would point to my server when on my local network, and now it's all good.   Woo!

**Reply by LumosX** [reply]
Just wanted to thank you for posting this. Adding the Upgrade and Connection headers solved my socket issues.
